### PR TITLE
51121 - Download ENC files from FSS for LargeMediaExchangSset

### PR DESCRIPTION
**Pull request by shailen14599**
Code changes & Unit Tests
Functional Tests

1. Removed temporary code for base folder creation 
2. Added temporary code for SCS BundleInfo generation to handle product location

### CODE REVIEW

**Review by harshal11869 — APPROVED** [review]
LGTM

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs [hunk 1]
```diff
@@ -112,15 +112,15 @@ private SalesCatalogueProductResponse GetSalesCatalogueResponse()
                                 EditionNumber = 2,
                                 UpdateNumbers = new List<int?> { 3, 4 },
                                  Dates = new List<Dates> {
-                            new Dates{ UpdateNumber= 4, UpdateApplicationDate= DateTime.Today, IssueDate = DateTime.Today}
-                        },
+                            new Dates{ UpdateNumber= 4, UpdateApplicationDate= DateTime.Today, IssueDate = DateTime.Today}},
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs:115]: The indentation is a bit off in this block - could you fix please

**Review by ajasonca — CHANGES_REQUESTED** [review]
Some minor issues to clean up

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 2]
```diff
@@ -107,22 +105,58 @@ public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenARea
...
+                var FolderName = $"M0{mediaNumber}X02";
+                bool checkFolder;
+                DownloadedFolderPath = await FileContentHelper.ExchangeSetLargeFile(ApiEssResponse, FssJwtToken, FolderName);
+                do
+                {
+                    string [] checkDirectories = FssBatchHelper.CheckforDirectories(Path.Combine(DownloadedFolderPath, $"B{baseNumber}", Config.ExchangeSetEncRootFolder));
+                    
+                    foreach (var folder in checkDirectories)
+                    {
+                        string[] checksubDirectories = FssBatchHelper.CheckforDirectories(Path.Combine(DownloadedFolderPath, $"B{baseNumber}", Config.ExchangeSetEncRootFolder, folder));
```
[the remark this answers is not in the document]
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated. Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 3]
```diff
@@ -107,22 +105,58 @@ public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenARea
...
+                        
+                        foreach (var updateFolder in checksubDirectories)
+                        {
+                            string[] checkupdateDirectories = FssBatchHelper.CheckforDirectories(Path.Combine(DownloadedFolderPath, $"B{baseNumber}", Config.ExchangeSetEncRootFolder, folder, updateFolder));
```
[the remark this answers is not in the document]
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated. Thanks

**Review by ajasonca — APPROVED** [review]
Looks good

**Pull request merged**